cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/sqlite/SQLDatabaseFactory.java

```diff
@@ -89,7 +90,9 @@ private static boolean isFtsAvailable() {
     public static SQLDatabase openSQLDatabase(File dbFile, KeyProvider provider) throws IOException, SQLException {
         Misc.checkNotNull(dbFile, "dbFile");
         File dbDirectory = dbFile.getParentFile();
-        dbDirectory.mkdirs();
+        if (!dbDirectory.mkdirs()){
+            logger.info(String.format(Locale.ENGLISH, "Did not create directories for path: %s directories may already exist", dbFile));
```

Comment:
I'd rather have a log message for a failure case than the successful case since the next two checks should work just fine. The level can always be lowered to `FINE`, `FINER` or `FINEST`.